.devcontainer/default/VSCode/settings.json

```diff
@@ -0,0 +1,59 @@
+{
+  // -- CMake Tools
+  "cmake.sourceDirectory": "${workspaceFolder}/llvm/llvm",
+  "cmake.buildDirectory": "${workspaceFolder}/build/circt",
+  "cmake.installPrefix": "/usr/local",
+  "cmake.configureOnOpen": false,
+  "cmake.configureOnEdit": false,
+  "cmake.generator": "Ninja",
+  "cmake.configureArgs": [
+    // CMAKE
+    "-DCMAKE_WARN_DEPRECATED=OFF",
+    "-DCMAKE_C_COMPILER_LAUNCHER=ccache",
+    "-DCMAKE_CXX_COMPILER_LAUNCHER=ccache",
+
+    // LLVM
+    "-DLLVM_ENABLE_ASSERTIONS=ON",
+    "-DLLVM_ENABLE_PROJECTS=mlir",
+    "-DLLVM_TARGETS_TO_BUILD=host",
+    "-DLLVM_BUILD_EXAMPLES=OFF",
+    "-DLLVM_ENABLE_OCAMLDOC=OFF",
+    "-DLLVM_ENABLE_BINDINGS=OFF",
+    "-DLLVM_ENABLE_MODULES=OFF",
+    "-DLLVM_USE_SANITIZER=",
+
+    // LLVM (performance)
+    "-DLLVM_USE_LINKER=lld-17",
+    "-DLLVM_OPTIMIZED_TABLEGEN=ON",
+    "-DLLVM_USE_SPLIT_DWARF=ON",
+
+    // CIRCT
+    "-DLLVM_EXTERNAL_PROJECTS=circt",
+    "-DLLVM_EXTERNAL_CIRCT_SOURCE_DIR=${workspaceFolder}",
+    "-DCIRCT_LLHD_SIM_ENABLED=OFF"
+  ],
+
+  // -- Clangd Settings
+  "clangd.path": "clangd-17",
+  "clangd.checkUpdates": false,
+  "clangd.onConfigChanged": "restart",
+  "clangd.arguments": [
+    "--all-scopes-completion",
+    "--compile-commands-dir=${workspaceFolder}/build/circt",
+    "--header-insertion=never",
+    "--clang-tidy",
+    "--clang-tidy-checks=*",
```

Comment:
Just curious, why all checks? (and not using the project's `.clang-tidy`?)

Reply:
The clang-tidy stuff is relatively new, I'll remove this line